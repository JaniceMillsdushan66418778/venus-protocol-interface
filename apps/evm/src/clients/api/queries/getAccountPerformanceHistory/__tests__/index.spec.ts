import { ChainId } from '@venusprotocol/chains';
import fakeAddress from '__mocks__/models/address';
import { restService } from 'utilities';
import { type Mock, vi } from 'vitest';
import { type ApiAccountPerformanceHistoryDataPoint, getAccountPerformanceHistory } from '..';

vi.mock('utilities/restService');

const fakeInput = {
  accountAddress: fakeAddress,
  chainId: ChainId.BSC_TESTNET,
  period: 'month' as const,
};

describe('getAccountPerformanceHistory', () => {
  it('returns correct sum for multiple destinationAmounts', async () => {
    (restService as Mock).mockImplementation(({ endpoint }: { endpoint: string }) => {
      const splitEndpoint = endpoint.split('/');
      const endpointType = splitEndpoint[splitEndpoint.length - 1];

      if (endpointType === 'performance') {
        const performanceDataPoints: ApiAccountPerformanceHistoryDataPoint[] = [
          {
            blockNumber: 1,
            blockTimestampMs: 1714828100000,
            netWorthCents: '10000',
          },
          {
            blockNumber: 3,
            blockTimestampMs: 1714828200000,
            netWorthCents: '11000',
          },
          {
            blockNumber: 5,
            blockTimestampMs: 1714828300000,
            netWorthCents: '9000',
          },
          {
            blockNumber: 7,
            blockTimestampMs: 1714828400000,
            netWorthCents: '8000',
          },
          {
            blockNumber: 9,
            blockTimestampMs: 1714828500000,
            netWorthCents: '20000',
          },
        ];

        return {
          data: {
            performanceDataPoints,
          },
        };
      }

      if (endpointType === 'net-worth') {
        const performanceDataPoints: ApiAccountPerformanceHistoryDataPoint[] = [
          {
            blockNumber: 6,
            blockTimestampMs: 1714828450000,
            netWorthCents: '8500',
          },
          {
            blockNumber: 10,
            blockTimestampMs: 1714828700000,
            netWorthCents: '21000',
          },
        ];

        return {
          data: {
            performanceDataPoints,
          },
        };
      }
    });

    const response = await getAccountPerformanceHistory(fakeInput);
    expect(response).toMatchSnapshot();
  });

  it('throws on error in payload', async () => {
    (restService as Mock).mockResolvedValue({ data: { error: 'Some error' } });

    await expect(getAccountPerformanceHistory(fakeInput)).rejects.toMatchObject({
      code: 'somethingWentWrong',
    });
  });

  it('throws on undefined payload', async () => {
    (restService as Mock).mockResolvedValue({ data: undefined });

    await expect(getAccountPerformanceHistory(fakeInput)).rejects.toMatchObject({
      code: 'somethingWentWrong',
    });
  });
});
