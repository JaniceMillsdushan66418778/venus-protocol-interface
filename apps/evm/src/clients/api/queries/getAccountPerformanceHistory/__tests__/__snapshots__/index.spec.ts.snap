// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getAccountPerformanceHistory > returns correct sum for multiple destinationAmounts 1`] = `
{
  "performanceHistory": [
    {
      "blockNumber": 1,
      "blockTimestampMs": 1714828100000,
      "netWorthCents": 10000,
    },
    {
      "blockNumber": 3,
      "blockTimestampMs": 1714828200000,
      "netWorthCents": 11000,
    },
    {
      "blockNumber": 5,
      "blockTimestampMs": 1714828300000,
      "netWorthCents": 9000,
    },
    {
      "blockNumber": 7,
      "blockTimestampMs": 1714828400000,
      "netWorthCents": 8000,
    },
    {
      "blockNumber": 9,
      "blockTimestampMs": 1714828500000,
      "netWorthCents": 20000,
    },
    {
      "blockNumber": 10,
      "blockTimestampMs": 1714828700000,
      "netWorthCents": 21000,
    },
  ],
  "startOfDayNetWorthCents": 8500,
}
`;
